packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,864 @@
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+
+import '../foundation/_features.dart';
+import 'binding.dart';
+import 'framework.dart';
+import 'view.dart';
+
+const String _windowingDisabledErrorMessage = '''
+Windowing APIs are not enabled.
+Windowing APIs are currently experimental. Do not use windowing APIs in
+production applications or plugins published to pub.dev.
+
+To try experimental windowing APIs:
+1. Switch to Flutter's main release channel.
+2. Turn on the windowing feature flag.
+
+See: https://github.com/flutter/flutter/issues/30701.
+''';
+
+/// Defines the possible archetypes for a window.
+///
+/// {@template flutter.widgets.windowing.experimental}
+/// Do not use this API in production applications or packages published to
+/// pub.dev. Flutter will make breaking changes to this API, even in patch
+/// versions.
+///
+/// This API throws an [UnsupportedError] error unless Flutter’s windowing
+/// feature is enabled by [isWindowingEnabled].
+///
+/// See: https://github.com/flutter/flutter/issues/30701.
+/// {@endtemplate}
+@internal
+enum WindowArchetype {
+  /// Defines a traditional window
+  ///
+  /// {@template flutter.widgets.windowing.experimental}
+  /// Do not use this API in production applications or packages published to
+  /// pub.dev. Flutter will make breaking changes to this API, even in patch
+  /// versions.
+  ///
+  /// This API throws an [UnsupportedError] error unless Flutter’s windowing
+  /// feature is enabled by [isWindowingEnabled].
+  ///
+  /// See: https://github.com/flutter/flutter/issues/30701.
+  /// {@endtemplate}
+  @internal
+  regular,
+}
+
+/// Defines sizing request for a window.
+///
+/// {@template flutter.widgets.windowing.experimental}
+/// Do not use this API in production applications or packages published to
+/// pub.dev. Flutter will make breaking changes to this API, even in patch
+/// versions.
+///
+/// This API throws an [UnsupportedError] error unless Flutter’s windowing
+/// feature is enabled by [isWindowingEnabled].
+///
+/// See: https://github.com/flutter/flutter/issues/30701.
+/// {@endtemplate}
+@internal
+class WindowSizing {
+  /// Creates a new [WindowSizing] object.
+  ///
+  /// {@template flutter.widgets.windowing.experimental}
+  /// Do not use this API in production applications or packages published to
+  /// pub.dev. Flutter will make breaking changes to this API, even in patch
+  /// versions.
+  ///
+  /// This API throws an [UnsupportedError] error unless Flutter’s windowing
+  /// feature is enabled by [isWindowingEnabled].
+  ///
+  /// See: https://github.com/flutter/flutter/issues/30701.
+  /// {@endtemplate}
+  @internal
+  WindowSizing({this.preferredSize, this.constraints}) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_windowingDisabledErrorMessage);
+    }
+  }
+
+  /// Preferred size of the window. This may not be honored by the platform.
```

Comment:
Google's technical writing recommendation is to prefer "might" over "may" except in legal contexts.

```suggestion
  /// Preferred size of the window. This might not be honored by the platform.
```